### ISSUE TITLE
On membership, in edit mode, on the edge between tablet & larger screens, there is a potential layout wobble

### ISSUE THREAD

**Issue by benjaminapetersen** [openshift/origin-web-console#1219]
With long or multiple roles, between ~765px and ~820px screen width there is the potential for this:
![screen shot 2017-02-03 at 11 35 03 am](https://cloud.githubusercontent.com/assets/280512/22599425/eaa6eaa2-ea04-11e6-9e83-c329be379250.png)

**Comment by spadgett** [comment]
Honestly I wonder if Bootstrap grid or even a table isn't better than flex for layouts like this.

**Comment by benjaminapetersen** [comment]
Agree, might be worth a restructure after all.

**Comment by benjaminapetersen** [comment]
Ideas:
- Could drop the 3rd column if we had a `+` action in the roles list that would open a dialog
- Could also eliminate edit/non-edit mode with the above `+` actions
- Or could list the role chips under the name, which would leave plenty of space (2 column again)

**Comment by sg00dwin** [comment]
There are various width/states, both edit and non-edit mode has content that is breaking the layout.

Adding examples:
<img width="322" alt="screen shot 2017-06-27 at 12 28 59 pm" src="https://user-images.githubusercontent.com/1874151/27600907-6dbb305e-5b3b-11e7-8ad5-85473666c8cd.png">
<img width="628" alt="screen shot 2017-06-27 at 12 28 11 pm" src="https://user-images.githubusercontent.com/1874151/27600903-6bcb199e-5b3b-11e7-89e3-24cd476234c2.png">

<img width="635" alt="screen shot 2017-06-27 at 1 17 14 pm" src="https://user-images.githubusercontent.com/1874151/27600914-7273497e-5b3b-11e7-8b2b-fbdce3f385ae.png">
<img width="852" alt="screen shot 2017-06-27 at 12 31 03 pm" src="https://user-images.githubusercontent.com/1874151/27600915-72784b7c-5b3b-11e7-8f6b-d36353c5038e.png">

**Comment by sg00dwin** [comment]
Posting some exploratory wireframes showing members and roles using PF list-view expand rows to address scalability and be consistent with other views.

View by member
![membership by user01](https://user-images.githubusercontent.com/1874151/28336438-ef7c05e2-6bcf-11e7-9211-8741547d0390.png)

View by member expanded
![membership by user expanded01](https://user-images.githubusercontent.com/1874151/28336439-ef7ea14e-6bcf-11e7-961e-d3b58e70e9b3.png)

Add member
![membership add user s 01](https://user-images.githubusercontent.com/1874151/28336441-ef84b408-6bcf-11e7-9626-aaccc7493b4a.png)

View by Role
![membership by role01](https://user-images.githubusercontent.com/1874151/28336457-025a5038-6bd0-11e7-8c64-5e7fa5f79857.png)

View by Role expanded
![membership by role expanded01](https://user-images.githubusercontent.com/1874151/28336456-0257ff04-6bd0-11e7-9e4d-7edf724d11d6.png)

cc @benjaminapetersen @jwforres @spadgett

**Comment by benjaminapetersen** [comment]
A quick hack @ the first image above could look something like this (keeping the tabs we have):

![screen shot 2017-07-18 at 4 24 12 pm](https://user-images.githubusercontent.com/280512/28338045-9c2c2ad8-6bd5-11e7-9a25-f2eeed357a54.png)

**Comment by beanh66** [comment]
@spadgett @benjaminapetersen I like what you have as a more immediate fix. Curious what it looks like when a row is expanded? In regards to the wireframes from @sg00dwin, I like the list view idea there too! As I mentioned we did something similar in the past with assigning roles to a list of names so here is a quick view of that functionality: http://g.recordit.co/GB9EJqbG9a.gif
In this case we used a table, but I could also see it in a list view instead. When there were too many roles to display on one row we just wrapped onto a second line so the rows got taller as needed. Not sure how we would incorporate the role descriptions but something to explore further.
